src/api/app/controllers/concerns/triggerable.rb

```diff
@@ -4,26 +4,26 @@ module Triggerable
   def set_project
     # By default we operate on the package association
     @project = @token.package.try(:project)
-    # If the token has no package, let's find one from the parameters
-    @project ||= Project.get_by_name(params[:project])
+    # If the token has no package, let's find one from the parameters or the step intructions
+    @project ||= Project.get_by_name(@project_name)
```

Comment:
:thought_balloon:  You could just call the methods that get those values instead of relying in instance variables that will be nil if someone forgets to fill them in. Calling methods that do not exist will throw a nice error that the programmer would notice faster. You could even write some default ones that throw `NotImplementedError` to make even more clear that they should be overriden.